sklearn/tree/_criterion.pyx

```diff
@@ -469,6 +675,43 @@ cdef class ClassificationCriterion(Criterion):
 
         self.pos = new_pos
 
+    cdef void update_missing_direction(self, SIZE_t new_dir) nogil:
+        """Update statistics by moving missing samples to the given direction.
+
+        Parameters
+        ----------
+        new_dir: SIZE_t
+             The new direction for the missing samples.
+             Note that irrespective of the direction, the missing samples
+             remain at the end.
+        """
+        cdef SIZE_t old_dir = self.missing_direction
+        self.missing_direction = new_dir
+        # TODO prevent updating the missing_direction if n_missing <= 0 ?
+        if old_dir == new_dir or self.n_missing <= 0:
+            return
+
+        cdef double* sum_missing = self.sum_missing
+        cdef double* sum_left = self.sum_left
+        cdef double* sum_right = self.sum_right
+        cdef SIZE_t* n_classes = self.n_classes
+
+        cdef SIZE_t k
+        cdef SIZE_t c
+
+        if new_dir == MISSING_DIR_LEFT:
+            # If missing must be moved from right to left
+            self.weighted_n_left += self.weighted_n_node_missing
+            self.weighted_n_right -= self.weighted_n_node_missing
+
+            for k in range(self.n_outputs):
+                for c in range(n_classes[k]):
+                    sum_left[c] += sum_missing[c]
+                    sum_right[c] -= sum_missing[c]
+                sum_missing += self.sum_stride
+                sum_left += self.sum_stride
+                sum_right += self.sum_stride
+
```

Comment:
+1 for renaming and removing the `new_dir` argument, since there is in fact only a single valid value